packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -319,30 +349,27 @@ export class FeatureDevController {
             })
             this.messenger.sendUpdatePlaceholder(tabID, 'Generating code ...')
             await session.send(message)
-            const filePaths = session.state.filePaths ?? []
-            const deletedFiles = session.state.deletedFiles ?? []
+
+            if (session.state.codeGenerationResult?.result === 'pending') {
+                throw new ToolkitError('Unexpected state in code generation')
+            }
+            if (session.state.codeGenerationResult?.result === 'failed') {
+                // enable the input if the error is not retryable.
+                // we're expecting a new prompt in this case
+                enableInput = !session.state.codeGenerationResult.retryable
```

Comment:
question: will there be failed code generation results that are non retryable? 

Only canned errors are at the moment handled and they are all non retryable.